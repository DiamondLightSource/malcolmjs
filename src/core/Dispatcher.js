/*
 * React.js Starter Kit
 * Copyright (c) 2014 Konstantin Tarkus (@koistya), KriaSoft LLC.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import Flux from 'flux';
import PayloadSources from '../constants/PayloadSources';
import assign from 'react/lib/Object.assign';

/**
 * A singleton that operates as the central hub for application updates.
 * For more information visit https://facebook.github.io/flux/
 */
let Dispatcher = assign(new Flux.Dispatcher(), {

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the server.
   */
  handleServerAction(action) {
    var payload = {
      source: PayloadSources.SERVER_ACTION,
      action: action
    };
    this.dispatch(payload);
  },

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the view.
   */
  handleViewAction(action) {
    var payload = {
      source: PayloadSources.VIEW_ACTION,
      action: action
    };
    this.dispatch(payload);
  }

});

export default Dispatcher;
