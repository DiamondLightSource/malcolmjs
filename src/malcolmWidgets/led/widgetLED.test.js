import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import blue from '@material-ui/core/colors/blue';
import WidgetLED from './widgetLED.component';

describe('WidgetLED', () => {
  let shallow;

  beforeEach(() => {
    shallow = createShallow({ dive: true });
  });

  it('renders off', () => {
    const wrapper = shallow(
      <WidgetLED
        LEDState={false}
        colorBorder={blue[300]}
        colorCenter={blue[700]}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders on', () => {
    const wrapper = shallow(
      <WidgetLED LEDState colorBorder={blue[300]} colorCenter={blue[700]} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
