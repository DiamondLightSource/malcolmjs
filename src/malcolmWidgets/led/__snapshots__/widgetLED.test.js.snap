// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetLED renders off 1`] = `
<div
  className="WidgetLED-div-1"
>
  <svg
    className="WidgetLED-svg-2"
    viewBox="0 0 48 48"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m36.25 8.6648c-7.1431-5.7237-17.356-5.7237-24.499 3e-7l-3.1781-3.1781-3.0856 3.0856 3.1781 3.1781c-5.7237 7.1431-5.7237 17.356 0 24.499l-3.1781 3.1781 3.0856 3.0856 3.1781-3.1781c7.1431 5.7237 17.356 5.7237 24.499 1e-6l3.1781 3.1781 3.0856-3.0856-3.1781-3.1781c5.7237-7.1431 5.7237-17.356 0-24.499l3.1781-3.1781-3.0856-3.0856-3.1781 3.1781zm-1.4502 26.135c-5.9628 5.9628-15.636 5.9628-21.599 0-5.9628-5.9628-5.9628-15.636 0-21.599 5.9628-5.9628 15.636-5.9628 21.599 0 5.9628 5.9628 5.9628 15.636 0 21.599zm8.706-12.981c-1.0036-9.0982-8.2255-16.32-17.324-17.324v-4.4945h-4.3636v4.4945c-9.0982 1.0036-16.32 8.2255-17.324 17.324h-4.4945v4.3636h4.4945c1.0036 9.0982 8.2255 16.32 17.324 17.324v4.4945h4.3636v-4.4945c9.0982-1.0036 16.32-8.2255 17.324-17.324h4.4945v-4.3636h-4.4945zm-19.505 17.455c-8.4327 0-15.273-6.84-15.273-15.273s6.84-15.273 15.273-15.273 15.273 6.84 15.273 15.273-6.84 15.273-15.273 15.273z"
      fill="#64b5f6"
    />
    ;
  </svg>
</div>
`;

exports[`WidgetLED renders on 1`] = `
<div
  className="WidgetLED-div-1"
>
  <svg
    className="WidgetLED-svg-2"
    viewBox="0 0 48 48"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m36.25 8.6648c-7.1431-5.7237-17.356-5.7237-24.499 3e-7l-3.1781-3.1781-3.0856 3.0856 3.1781 3.1781c-5.7237 7.1431-5.7237 17.356 0 24.499l-3.1781 3.1781 3.0856 3.0856 3.1781-3.1781c7.1431 5.7237 17.356 5.7237 24.499 1e-6l3.1781 3.1781 3.0856-3.0856-3.1781-3.1781c5.7237-7.1431 5.7237-17.356 0-24.499l3.1781-3.1781-3.0856-3.0856-3.1781 3.1781zm-1.4502 26.135c-5.9628 5.9628-15.636 5.9628-21.599 0-5.9628-5.9628-5.9628-15.636 0-21.599 5.9628-5.9628 15.636-5.9628 21.599 0 5.9628 5.9628 5.9628 15.636 0 21.599zm8.706-12.981c-1.0036-9.0982-8.2255-16.32-17.324-17.324v-4.4945h-4.3636v4.4945c-9.0982 1.0036-16.32 8.2255-17.324 17.324h-4.4945v4.3636h4.4945c1.0036 9.0982 8.2255 16.32 17.324 17.324v4.4945h4.3636v-4.4945c9.0982-1.0036 16.32-8.2255 17.324-17.324h4.4945v-4.3636h-4.4945zm-19.505 17.455c-8.4327 0-15.273-6.84-15.273-15.273s6.84-15.273 15.273-15.273 15.273 6.84 15.273 15.273-6.84 15.273-15.273 15.273z"
      fill="#64b5f6"
    />
    ;
    <circle
      cx="24.5"
      cy="24.5"
      fill="#1976d2"
      r="12"
    />
  </svg>
</div>
`;
